snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -569,8 +570,11 @@ def mark_build_done(self):
             scriptlet_metadata=self._scriptlet_metadata[steps.BUILD]))
 
     def _get_machine_manifest(self):
+        # Use subprocess directly here. common.run_output will use binaries out
+        # of the snap, and we want to use the one on the host.
+        uname = subprocess.check_output(['uname', '-srvmpio'])
```

Comment:
We will need a specific error for this now (it was catched by the general wrapper until this change).